### ISSUE TITLE
`briefcase package` is missing to sign long with libraries capabilities when required

### ISSUE THREAD

**Issue by kushaldas** [beeware/briefcase#652]
**Describe the bug**

While building [Tumpa](https://tumpa.rocks) I found that created dmg is passing notarization but does start on any other Mac, saying "Developer cannot be verified". 
After fighting with Apple notarization system for many hours, I found that while running the `codesign` on the files, I will have to pass `library` as option.

I had to patch `briefcase/platforms/macOS/__init__.py` the `sign` function to have the following:

```diff
--- .venv/lib/python3.9/site-packages/briefcase/platforms/macOS/__init__.py	2022-01-07 08:48:12.000000000 +0100
+++ /tmp/__init__.py	2022-01-07 08:47:54.000000000 +0100
@@ -117,7 +117,7 @@
                     '--deep', str(path),
                     '--force',
                     '--timestamp',
-                    '--options', 'runtime',
+                    '--options', 'runtime,library',
                 ],
                 check=True,
             )
```

**Environment:**
 - Operating System: macOS Monterey, 12.1
 - Python version: Python 3.9.1
 - Software versions:
   - Briefcase: 0.3.5

**Comment by freakboy3742** [comment]
Hrm - this is interesting; the docs for `codesign` say:

> **library**  Forces the signed code's library validation flag to be set when the code begins execution.  The code will only be able to link against system libraries and frameworks, or libraries, frameworks, and plug-in bundles with the same team identifier embedded in the code directory.  Team identifiers are automatically recorded in signatures when signing with suitable Apple-issued signing certificates.  Note that the flag is not supported for i386 binaries, and only applies to the main executable.  The flag has no effect when set on frameworks and libraries.
>
> **runtime** On macOS versions >= 10.14.0, opts signed processes into a hardened runtime environment which includes runtime code signing enforcement, library validation, hard, kill, and debugging restrictions.  These restrictions can be selectively relaxed via entitlements. Note: macOS versions older than 10.14.0 ignore the presence of this flag in the code signature.

Based on that, it would make sense to me if you saw this bug on High Sierra (10.13); but on Monterey, I read that as `runtime` infers `library`. 

A couple of follow up questions:
1. Are you running this on an M1 machine, or an x86 machine? Some of the other options look like they might have 
2. What can you tell me about Tumpa itself? I can see in the repo that it's a Pyside2 app with a binary wheel; is there anything else of note about the
3. Does `codesign --verify` reveal anything about the app signature when it is in it's "broken" state?

**Comment by dreavjr** [comment]
I, too, found the same issue. I was getting a stubborn GateKeeper error:
```
ImportError: dlopen(.../QtGui.abi3.so, 2): Library not loaded:.../QtGui.abi3.so
  Reason: no suitable image found.  Did find:
	.../QtGui: code signature in (.../QtGui) not valid for use in process using Library Validation: mapped file has no cdhash, completely unsigned? Code has to be at least ad-hoc signed.
```

I tried everything (not signing, signing ad-hoc, signing with an Apple Developer Certificate) but the only solution was applying the patch recommended by @kushaldas (thanks!).

Strangely, `codesign --verify` accuses nothing ("valid on disk", "satisfies its Designated Requirement"), both for the bundle and for that particular file.

- Operating System: macOS Big Sur 11.6.2
- Python version: Python 3.8.9
- Briefcase: 0.3.5
- Hardware: iMac 2017 (x86)

**Comment by freakboy3742** [comment]
@dreavjr Did you have any luck with your procedure? Even if you didn't have complete success, any details you can share would be helpful.

**Comment by dreavjr** [comment]
@freakboy3742 Yes, I was successful, but the changes I had to implement in the patch were not exactly small. I will prepare a pull request and let you judge what you are willing to incorporate in the main branch,

**Comment by freakboy3742** [comment]
@dreavjr I'll take a large, complex but functional patch over a small broken one any day :-)

**Comment by freakboy3742** [comment]
@kushaldas Are you able to confirm whether #690 resolves your packaging issues with tumba? I tried to package it myself, but got caught on the johnnycanencrypt wheel.

**Comment by kushaldas** [comment]
> @kushaldas Are you able to confirm whether #690 resolves your packaging issues with tumba? I tried to package it myself, but got caught on the johnnycanencrypt wheel.

I will give it a try this week, I somehow miss all the github notifications.

**Comment by freakboy3742** [comment]
@kushaldas I've merged #690 on the basis that all my testing reveals that it is at least an improvement on the old signing; if you have any problems with packaging rumba, we can reopen this ticket.
